src/elliptic/curves/traits.rs

```diff
@@ -139,11 +144,11 @@ pub trait ECPoint: Zeroize + Clone + PartialEq + fmt::Debug + 'static {
     /// Returns error if x, y are not on curve
     fn from_coords(x: &BigInt, y: &BigInt) -> Result<Self, NotOnCurve>;
     /// Returns `x` coordinate of the point, or `None` if point is at infinity
-    fn x_coord(&self) -> Option<BigInt>;
+    fn x_coord(&self) -> BigInt;
     /// Returns `y` coordinate of the point, or `None` if point is at infinity
-    fn y_coord(&self) -> Option<BigInt>;
+    fn y_coord(&self) -> BigInt;
     /// Returns point coordinates (`x` and `y`), or `None` if point is at infinity
-    fn coords(&self) -> Option<PointCoords>;
+    fn coords(&self) -> PointCoords;
```

Comment:
In most curvs we returned (0,0) at points at infinity anywhere, only in secp and bls we returned None, so this is my attempt on making it consistent

Reply:
We don't (didn't) return (0,0) at points at infinity anywhere — points at infinity don't have coordinates. We return (x0,y0) for ed25519 and ristretto, because their neutral element has coordinates (x0,y0). Edwards curves has no point at infinity because they are "complete", but Weierstrass and Montgomery curves needs a point at infinity to represent identity point. As we support all these types of curves, we need to take this fact, that certain points might not have coordinates, into account.

Reply:
Hmm I see, so in ed25519/ristretto we have coordinates for the identity element but in weiestrass/montgomery curves it's the imaginative inifinity point.
that makes sense :)